cmds/cmd.c

```diff
@@ -83,15 +83,24 @@ void cmd_reg(const cmd_t *cmd)
 	unsigned int i;
 
 	for (i = 0; i < SIZE_CMDS; ++i) {
-		if (cmd_common.cmds[i] != NULL) {
-			if (hal_strcmp(cmd_common.cmds[i]->name, cmd->name) == 0)
-				return;
+		if (cmd_common.cmds[i] == NULL) {
+			cmd_common.cmds[i] = cmd;
+			return;
+		}
 
-			continue;
+		if (hal_strcmp(cmd_common.cmds[i]->name, cmd->name) == 0) {
+			break;
 		}
+	}
 
-		cmd_common.cmds[i] = cmd;
-		break;
+	/*
+	 * FIXME: Oops, the command is already registered or no free slots
```

Comment:
Generic comment - might be done in the future:
as we do have linker scripts now, we could probably just put all `cmd_t` structs in special section, remove constructors and const SIZE_CMDS altogether (commands would be "registered" by `ld` by placing them in a section with extra symbols at the beginning/end so we can iterate over them). 

This would allow us to avoid problems like that in the future (and remove the runtime overhead of registering CMDs, we could even sort commands and to bsearch if we would like to be fancy ;)).

Reply:
Thanks, it's a good idea, I'll probably do it, but in a separate PR, after refactoring the devices. However, now I will add a comment in the code, so that I don't miss this idea.